Part4

hobwell opened the pull request.

hobwell committed:
Steps 1 to 7

Moved all struct declarations to .h files
Move all struct implementations to .cpp files
Moved nested structs into their own .h and .cpp files
Added necessary includes
Updated .replit file

hobwell committed:
Step 8

Marked function parameters as const, where applicable

hobwell committed:
Fixed formatting

Fixed formatting
re-ordered ValueDisplay methods in cpp file

matkatmusic commented on TrashBin.h:
```diff
@@ -0,0 +1,25 @@
+
+struct TrashBin
+{
+    TrashBin (Cat& nearbyCat);
```
Why does this `Cat` need to be a reference and not a `const &`? 
The code in the `TrashBin` constructor does not modify the instance.
- [x] investigate and revise if possible.

hobwell replied:
It can be const.  When I read "function implementations" I did not mentally include the constructors, for some reason.  I've gone back through and updated all constructors.

matkatmusic commented on Wrappers.h:
```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include "Cat.h"
+#include "Fruit.h"
+#include "Person.h"
+#include "TrashBin.h"
+#include "Treadmill.h"
```
Don't include what you don't use.  
All of these types can be forward declared here.  include these files in the CPP file.
- [x] revise

hobwell replied:
Fixed, thank you.

hobwell committed:
Resolved issues

forward declared structs in Wrappers.h
added includes to Wrappers.cpp
updated UDT constructors with const arguments, where applicable.
